### PR TITLE
ADS: Widgets migration

### PR DESCRIPTION
**Pull request by nalcalag**
Task/Issue URL: 

### Description

Migrate widgets to new styles

### Steps to test this PR

- Install from this branch
- Add widget to your home screen
- Check widgets match new designs

### UI changes
| Before  | After |
| ------ | ----- |
Screenshot_20221230_163631_One UI Home|Screenshot_20221230_163636_One UI Home|

### CODE REVIEW

**Comment by nalcalag** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2668** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2668" target="_blank">Graphite</a> 
    * **PR #2687** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2687" target="_blank">Graphite</a>  👈
    * **PR #2682** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2682" target="_blank">Graphite</a> 
    * **PR #2678** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2678" target="_blank">Graphite</a> 
    * **PR #2677** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2677" target="_blank">Graphite</a> 
    * **PR #2666** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2666" target="_blank">Graphite</a> 
      * **PR #2667** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2667" target="_blank">Graphite</a> 
      * **PR #2671** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2671" target="_blank">Graphite</a> 
    * **PR #2674** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2674" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2687?utm_source=stack-comment).

File: app/src/main/res/drawable/ic_find_search_20_daynight.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  -->
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="20dp"
+    android:height="20dp"
+    android:viewportWidth="20"
+    android:viewportHeight="20">
+  <path
+      android:pathData="M9,2C5.134,2 2,5.134 2,9s3.134,7 7,7c1.645,0 3.157,-0.567 4.352,-1.517l3.282,3.283c0.313,0.312 0.82,0.312 1.132,0 0.312,-0.313 0.312,-0.82 0,-1.132l-3.283,-3.282C15.433,12.157 16,10.645 16,9c0,-3.866 -3.134,-7 -7,-7zM3.6,9c0,-2.982 2.418,-5.4 5.4,-5.4s5.4,2.418 5.4,5.4 -2.418,5.4 -5.4,5.4S3.6,11.982 3.6,9z"
+      android:fillColor="@color/searchWidgetSearchBarTrailingIconColor"
```
Comment (malmstein) [inline app/src/main/res/drawable/ic_find_search_20_daynight.xml:24]: can we use a daxColor value for this?

File: app/src/main/res/drawable/ic_find_search_20_light.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  -->
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="20dp"
+    android:height="20dp"
+    android:viewportWidth="20"
+    android:viewportHeight="20">
+  <path
+      android:pathData="M9,2C5.134,2 2,5.134 2,9s3.134,7 7,7c1.645,0 3.157,-0.567 4.352,-1.517l3.282,3.283c0.313,0.312 0.82,0.312 1.132,0 0.312,-0.313 0.312,-0.82 0,-1.132l-3.283,-3.282C15.433,12.157 16,10.645 16,9c0,-3.866 -3.134,-7 -7,-7zM3.6,9c0,-2.982 2.418,-5.4 5.4,-5.4s5.4,2.418 5.4,5.4 -2.418,5.4 -5.4,5.4S3.6,11.982 3.6,9z"
+      android:fillColor="@color/black84"
```
Comment (malmstein) [inline app/src/main/res/drawable/ic_find_search_20_light.xml:24]: ditto

File: app/src/main/res/drawable/ic_find_search_20_dark.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  -->
+
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="20dp"
+    android:height="20dp"
+    android:viewportWidth="20"
+    android:viewportHeight="20">
+  <path
+      android:pathData="M9,2C5.134,2 2,5.134 2,9s3.134,7 7,7c1.645,0 3.157,-0.567 4.352,-1.517l3.282,3.283c0.313,0.312 0.82,0.312 1.132,0 0.312,-0.313 0.312,-0.82 0,-1.132l-3.283,-3.282C15.433,12.157 16,10.645 16,9c0,-3.866 -3.134,-7 -7,-7zM3.6,9c0,-2.982 2.418,-5.4 5.4,-5.4s5.4,2.418 5.4,5.4 -2.418,5.4 -5.4,5.4S3.6,11.982 3.6,9z"
+      android:fillColor="@color/white84"
```
Comment (malmstein) [inline app/src/main/res/drawable/ic_find_search_20_dark.xml:24]: ditto

File: app/src/main/res/drawable/search_widget_background_black.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="@color/gray90" />
```
Comment (malmstein) [inline app/src/main/res/drawable/search_widget_background_black.xml:18]: ditto

File: app/src/main/res/drawable/search_widget_background_white.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="@color/white" />
```
Comment (malmstein) [inline app/src/main/res/drawable/search_widget_background_white.xml:18]:
```suggestion
    <solid android:color="?attr/daxColorWhite" />
```

File: app/src/main/res/drawable/search_widget_favorite_favicon_dark_background.xml [hunk 6]
```diff
@@ -15,6 +15,6 @@
   -->
 
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
-    <solid android:color="@color/almostBlackDark" />
+    <solid android:color="@color/white12" />
     <corners android:radius="10dp" />
```
Comment (malmstein) [inline app/src/main/res/drawable/search_widget_favorite_favicon_dark_background.xml:19]: is this 10dp? can we use an already set radius?

File: app/src/main/res/drawable/search_widget_favorite_favicon_light_background.xml [hunk 7]
```diff
@@ -15,6 +15,6 @@
   -->
 
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
-    <solid android:color="@color/whiteFive" />
+    <solid android:color="@color/black6" />
     <corners android:radius="10dp" />
```
Comment (malmstein) [inline app/src/main/res/drawable/search_widget_favorite_favicon_light_background.xml:19]: is this 10dp? can we use an already set radius?

File: app/src/main/res/drawable/search_widget_favorites_dark_background.xml [hunk 8]
```diff
@@ -15,6 +15,6 @@
   -->
 
 <shape xmlns:android="http://schemas.android.com/apk/res/android">
-    <solid android:color="@color/newBlack" />
+    <solid android:color="@color/gray90" />
     <corners android:radius="20dp" />
```
Comment (malmstein) [inline app/src/main/res/drawable/search_widget_favorites_dark_background.xml:19]: is this 20dp? can we use an already set radius?

File: common-ui/src/main/res/values-night/widget-colors.xml [hunk 9]
```diff
@@ -33,10 +33,12 @@
 
 <resources>
     <!-- Widget colors -->
-    <color name="favoriteTitleTextColor">@color/white</color>
-    <color name="favoriteFaviconBackgroundColor">@color/almostBlackDark</color>
-    <color name="searchWidgetBackgroundColor">@color/newBlack</color>
-    <color name="searchWidgetSearchBarBackgroundColor">@color/almostBlack</color>
-    <color name="searchWidgetSearchBarHintColor">@color/white</color>
-    <color name="searchWidgetSearchBarLoupeColor">@color/white</color>
+    <color name="favoriteTitleTextColor">@color/white60</color>
```
Comment (malmstein) [inline common-ui/src/main/res/values-night/widget-colors.xml:36]: this is great, can we do this for all the different @color used in the resources?

File: voice-search/voice-search-impl/src/main/res/drawable/ic_microphone_light.xml [hunk 10]
```diff
@@ -20,9 +20,9 @@
     android:viewportWidth="24"
     android:viewportHeight="24">
     <path
-        android:fillColor="@color/almostBlack"
+        android:fillColor="@color/black84"
```
Comment (malmstein) [inline voice-search/voice-search-impl/src/main/res/drawable/ic_microphone_light.xml:23]: this should be using attr colors

**Review by malmstein — CHANGES_REQUESTED** [review]
There are loads of colors used that should be using attr values.

File: app/src/main/res/drawable/ic_find_search_20_daynight.xml (hunk 1 above)
Comment (nalcalag) [reply app/src/main/res/drawable/ic_find_search_20_daynight.xml]: You can't use custom `attrs` from the theme in Widgets. Therefore it needs to be like the existing configuration.

File: app/src/main/res/drawable/ic_find_search_20_light.xml (hunk 2 above)
Comment (nalcalag) [reply app/src/main/res/drawable/ic_find_search_20_light.xml]: same as previous comment

File: app/src/main/res/drawable/ic_find_search_20_dark.xml (hunk 3 above)
Comment (nalcalag) [reply app/src/main/res/drawable/ic_find_search_20_dark.xml]: Can't use daxColors in widgets

File: app/src/main/res/drawable/search_widget_background_white.xml (hunk 5 above)
Comment (nalcalag) [reply app/src/main/res/drawable/search_widget_background_white.xml:18]: Can't use daxColors in widgets

File: app/src/main/res/drawable/search_widget_background_black.xml (hunk 4 above)
Comment (nalcalag) [reply app/src/main/res/drawable/search_widget_background_black.xml:18]: Can't use daxColors in widgets

File: voice-search/voice-search-impl/src/main/res/drawable/ic_microphone_light.xml (hunk 10 above)
Comment (nalcalag) [reply voice-search/voice-search-impl/src/main/res/drawable/ic_microphone_light.xml:23]: Can't use daxColors in widgets

File: app/src/main/res/drawable/search_widget_favorite_favicon_dark_background.xml (hunk 6 above)
Comment (nalcalag) [reply app/src/main/res/drawable/search_widget_favorite_favicon_dark_background.xml:19]: There is not any set radius with value `10dp` (S)8dp/(M)12dp/(L)16dp

File: app/src/main/res/drawable/search_widget_favorite_favicon_light_background.xml (hunk 7 above)
Comment (nalcalag) [reply app/src/main/res/drawable/search_widget_favorite_favicon_light_background.xml:19]: ditto

File: common-ui/src/main/res/values-night/widget-colors.xml (hunk 9 above)
Comment (nalcalag) [reply common-ui/src/main/res/values-night/widget-colors.xml:36]: I removed the ones we are no using anymore after removing them from widgets. 
I suppose we can remove all resources in a different task. I wouldn't like to maybe add some bugs here after design review is done for this task.

File: app/src/main/res/drawable/search_widget_favorites_dark_background.xml (hunk 8 above)
Comment (nalcalag) [reply app/src/main/res/drawable/search_widget_favorites_dark_background.xml:19]: We agreed to change it to 16dp 👍

File: app/src/main/res/drawable/search_widget_favorite_favicon_dark_background.xml (hunk 6 above)
Comment (nalcalag) [reply app/src/main/res/drawable/search_widget_favorite_favicon_dark_background.xml:19]: We agreed to change it to 8dp

**Comment by nalcalag** [comment]
Follow up in https://github.com/duckduckgo/Android/pull/2711

**Pull request closed without merging**